### PR TITLE
Remove trailing whitespace from docs files

### PR DESCRIPTION
**Pull request by awly**
Some editors will do this automatically on save. This causes a lot of
diffs when editing the docs in such an editor.
Clean them up once now and we'll try to keep it tidy going forward.

### CODE REVIEW

**Review by webvictim — APPROVED** [review]
LGTM. 

Is there a command or one-liner you would recommend using to do this automatically?

I recently installed an extension in my editor (VS Code) which highlights trailing whitespace too: https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces

**Review by benarent — APPROVED** [review]
👍 but please don't merge yet, I'm going to do some other plumping to the 4.3 branch today and I know in the past it's got me cherry picking. I'll give you final thumbs up today.

**Comment by benarent** [comment]
^ Ok, we are good to go. You might want to rebase and this should resolve the conflicts.

**Comment by awly** [comment]
> Is there a command or one-liner you would recommend using to do this automatically?

This does the trick for a directory:
```sh
find docs/4.3 -type f | xargs sed -i -E 's/\s+$//g'
```

Should we do it on `make docs` automatically for all versions?

**Comment by webvictim** [comment]
> Should we do it on `make docs` automatically for all versions?

That sounds pretty good to me, what @benarent thinks is more relevant though as he does more regular changes. It'd be nice if we could do it as the same time as we run `milv` to validate the links.

In the new Drone CI world we can probably have a conditional job that runs on `docs/**` PRs and fails if there's extra whitespace.

**Comment by benarent** [comment]
Thanks gents, I'm always open to improving my whitespace game.. I'm the likely culprit for adding them. 

Let's add this into a make file, that'll run Milv and cleanup whitespace at the same time.

**Comment by awly** [comment]
ok, added make rules for testing/fixing whitespace and milv.
`make docs` will test for whitespace and run milv.
`make docs-fix-whitespace` fixes trailing whitespace.
we can add `make docs-test` into our Jenkins config too.

PTAL @benarent @webvictim

File: Makefile
Comment (benarent) [inline Makefile:213]: 👍 very handy input.

File: Makefile
Comment (benarent) [inline Makefile:225]: When we introduced Milv we didn't run it on older versions, since now moved out older version and only have current the mass milv fix should be must easier. Once merged, I'll fix the rest of the output.

**Review by benarent — APPROVED** [review]

**Pull request merged**
